features/release.feature

```diff
@@ -367,6 +367,36 @@ MARKDOWN
           ASSET_TARBALL
           """
 
+  Scenario: Delete a release
+    Given the GitHub API server:
+      """
+      get('/repos/mislav/will_paginate/releases') {
+          json [
+            { url: 'https://api.github.com/repos/mislav/will_paginate/releases/123',
+              upload_url: 'https://api.github.com/uploads/assets{?name,label}',
+              tag_name: 'v1.2.0',
+              name: 'will_paginate 1.2.0',
+              draft: true,
+              prerelease: false,
+              assets: [
+                { url: 'https://api.github.com/repos/mislav/will_paginate/assets/9876',
+                  name: 'hello-1.2.0.tar.gz',
+                },
+              ],
+            },
+          ]
+      }
+
+      delete('/repos/mislav/will_paginate/releases/123') {
+        status 204
+      }
+      """
+    When I successfully run `hub release delete v1.2.0`
+    Then the output should contain exactly:
+      """
+      Deleting release related to v1.2.0...\n
```

Comment:
When you get rid of this output, as advised in my previous comment, then it's going to be hard to assert if the command actually did anything.

If you're comfortable with Ruby, you can amend the api server to keep the list of releases in memory while it's running. Then you can issue several requests to it. For example:

```rb
releases = [
  { tag_name: 'v1.2.0',
    id: '...',
    url: '...
  },
  { tag_name: 'v1.2.1',
    id: '...',
    url: '...
  },
]

get('/repos/mislav/will_paginate/releases') {
  json releases
}

delete('/repos/mislav/will_paginate/releases/:id') {
  if releases.reject! { |r| r.id.to_s == params[:id] }
    status 204
  else
    status 404
  end
}
```

Then you can do `hub release delete v1.2.0`, verify that the output is blank, then do `hub release` and verify that "v1.2.0" no longer appears in the output. Or, you could try to run `hub release delete ...` twice and verify that the 2nd run fails. What do you think?